roachpb/data.go

```diff
@@ -546,6 +546,9 @@ func (t *Transaction) Update(o *Transaction) {
 		*t = *proto.Clone(o).(*Transaction)
 		return
 	}
+	if len(t.Key) == 0 {
+		t.Key = o.Key
+	}
 	if o.Status != PENDING {
 		t.Status = o.Status
 	}
```

Comment:
I re-enabled the assertion at for status regression, but I don't see an earlier TODO you're referring to.